src/psyclone/dynamo0p3_builtins.py

```diff
@@ -213,6 +213,8 @@ def __str__(self):
         return "Built-in: Add fields"
 
     def gen_code(self, parent):
+        ''' Generates Dynamo0.3 API specific PSy code for a call to the
+            X_plus_Y Built-in. '''
```

Comment:
Thanks for the review, @arporter. I am currently working on changes so I would like to check whether I got things right here.
- As a built-in call is essentially looping over DoFs so the parent type should be `psyclone.f2pygen.BaseGen`
- As for the `:param parent:` description, I noticed that in `dynamo0p3.py` the `psyclone.f2pygen.BaseGen` (in `gen_code`) docstrings were 
`        
:param parent: an f2pygen object that will be the parent of
 f2pygen objects created in this method
`
A call to built-in can be a part of a larger invoke object and it can also stand on its own. Does the `Node in f2pygen to which to add call` have similar meaning to the one above?

Reply:
At the level of f2pygen, we are dealing only with Fortran code - there's no relation back to PSyclone-type concepts such as looping over DoFs. `f2pygen.BaseGen` is the base class for all objects used to create Fortran and thus is the correct type for the parent.
Yes `Node in f2pygen to which to add call` is still correct - when generating Fortran the built-in will always be a child of something (e.g. SubroutineGen). Even when generating a kernel stub (which doesn't apply for built-ins), the routine will be a child of a module. It might be better to have `Node in f2pygen tree to which to ...` to emphasise that we are building a tree.

Reply:
Thank you very much for the clarification. In the end I went with:

        :param parent: Node in f2pygen tree to which to add call
        :type parent: :py:class:`psyclone.f2pygen.BaseGen`

Reply:
Thank you for the clarification. In the end I went with e.g.

        '''
        Generates Dynamo0.3 API specific PSy code for a call to the
        X_plus_Y Built-in.

        :param parent: Node in f2pygen tree to which to add call
        :type parent: :py:class:`psyclone.f2pygen.BaseGen`
        '''